src/api/erc20/Erc20ApiProxy.ts

```diff
@@ -0,0 +1,58 @@
+import BN from 'bn.js'
+
+import { Receipt } from 'types'
+
+import {
+  Erc20Api,
+  NameParams,
+  SymbolParams,
+  DecimalsParams,
+  TotalSupplyParams,
+  BalanceOfParams,
+  AllowanceParams,
+  ApproveParams,
+  TransferParams,
+  TransferFromParams,
+} from './Erc20Api'
+import { CacheProxy } from 'api/proxy'
+
+const DEFAULT_CACHE_TIME = 30
+
+export class Erc20ApiProxy extends CacheProxy<Erc20Api> implements Erc20Api {
```

Comment:
> Are you sure we want to use extension instead of composition?

Not really. This is the approach that made more sense at the time, but ok trying a different way.

Reply:
I'm happy either way. I'm kind of thinking aloud

Reply:
I tried a different approach in the latest version. Take a look if you can, otherwise you can give your opinion when you get back from vacation :)